personalities/ct-static-api/config.go

```diff
@@ -196,159 +149,29 @@ func ValidateLogConfig(cfg *configpb.LogConfig) (*ValidatedLogConfig, error) {
-			return nil, fmt.Errorf("invalid frozen STH: %v", err)
-		}
-		if err := verifier.VerifySTHSignature(*vCfg.FrozenSTH); err != nil {
-			return nil, fmt.Errorf("signature verification failed: %v", err)
-		}
-	}
-
-	switch cfg.ExtraDataIssuanceChainStorageBackend {
-	case configpb.LogConfig_ISSUANCE_CHAIN_STORAGE_BACKEND_CTFE:
-		// Validate the combination of CtfeStorageConnectionString and ExtraDataIssuanceChainStorageBackend
-		if len(cfg.CtfeStorageConnectionString) == 0 {
-			return nil, errors.New("missing ctfe_storage_connection_string when issuance chain storage backend is CTFE")
-		}
-		// Validate CTFEStorageConnectionString
-		if !strings.HasPrefix(cfg.CtfeStorageConnectionString, "mysql") {
-			return nil, errors.New("unsupported driver in ctfe_storage_connection_string")
-		}
-		if _, err := mysql.ParseDSN(strings.Split(cfg.CtfeStorageConnectionString, "://")[1]); err != nil {
-			return nil, errors.New("failed to parse ctfe_storage_connection_string for mysql driver")
-		}
-		vCfg.CTFEStorageConnectionString = cfg.CtfeStorageConnectionString
-	case configpb.LogConfig_ISSUANCE_CHAIN_STORAGE_BACKEND_TRILLIAN_GRPC:
-		// Nothing to validate for Trillian gRPC
-	}
-	vCfg.ExtraDataIssuanceChainStorageBackend = cfg.ExtraDataIssuanceChainStorageBackend
-
 	return &vCfg, nil
 }
 
-// LogBackendMap is a map from log backend names to LogBackend objects.
-type LogBackendMap = map[string]*configpb.LogBackend
-
-// BuildLogBackendMap returns a map from log backend names to the corresponding
-// LogBackend objects. It returns an error unless all backends have unique
-// non-empty names and specifications.
-func BuildLogBackendMap(lbs *configpb.LogBackendSet) (LogBackendMap, error) {
-	lbm := make(LogBackendMap)
-	specs := make(map[string]bool)
-	for _, be := range lbs.Backend {
-		if len(be.Name) == 0 {
-			return nil, fmt.Errorf("empty backend name: %v", be)
-		}
-		if len(be.BackendSpec) == 0 {
-			return nil, fmt.Errorf("empty backend spec: %v", be)
-		}
-		if _, ok := lbm[be.Name]; ok {
-			return nil, fmt.Errorf("duplicate backend name: %v", be)
-		}
-		if ok := specs[be.BackendSpec]; ok {
-			return nil, fmt.Errorf("duplicate backend spec: %v", be)
-		}
-		lbm[be.Name] = be
-		specs[be.BackendSpec] = true
-	}
-	return lbm, nil
-}
-
-func validateConfigs(cfg []*configpb.LogConfig) error {
+func ValidateLogConfigSet(cfg *configpb.LogConfigSet) error {
 	// Check that logs have no duplicate or empty prefixes. Apply other LogConfig
 	// specific checks.
 	logNameMap := make(map[string]bool)
-	for _, logCfg := range cfg {
+	for _, logCfg := range cfg.Config {
 		if _, err := ValidateLogConfig(logCfg); err != nil {
 			return fmt.Errorf("log config: %v: %v", err, logCfg)
 		}
-		if len(logCfg.Prefix) == 0 {
-			return fmt.Errorf("log config: empty prefix: %v", logCfg)
-		}
-		if logNameMap[logCfg.Prefix] {
-			return fmt.Errorf("log config: duplicate prefix: %s: %v", logCfg.Prefix, logCfg)
+		if len(logCfg.Origin) == 0 {
```

Comment:
I think this is checked in `ValidateLogConfig`?

Reply:
Yeah the code in config.go was unnecessarily complex I think. Probably because the multiconfig stuff was added later on? I've simplified it a bunch, ptal.